cli/internal/helm/client.go

```diff
@@ -116,6 +119,10 @@ func (c *Client) upgradeRelease(
 		return nil
 	}
 
+	if releaseName == certManagerReleaseName && !confirmation {
+		return ErrConfirmationMissing
+	}
```

Comment:
Maybe add a unit test to cover this case and check the correct error is returned.